### PR TITLE
test(rpc_test): :white_check_mark: Added support for test fixtures + refactor

### PR DESCRIPTION
**Pull request by Trantorian1**
✅ **fixtures added**: all unit tests now have access to fixtures defined in `unit_tests/src/fixtures.rs`, which are functions that define how to produce dependencies.
♻️ **RPC client refactoring**: RPC clients are now accessible through a HashMap generated in a fixture. *Please use only the constants defined in `unit_tests/src/constants.rs` to access them*.

eg:
```rs
#[rstest]
#[tokio::test]
async fn my_unit_test(clients: HashMap<String, JsonRpcClient<HttpTransport>>) {
  let deoxys = &clients[DEOXYS]
  // ...
}
```

🚚 **test crate renaming**: this was due to conflicts with cargo's naming scheme that would generate bugs when compiling / runnnin unit tests. Please migrate your tests to `unit_tests` as soon as possible.
♻️ **refactoring of imports and constants**: all test contants are now in `unit_tests/src/constants.rs`. A `unit_tests/tests/common.rs` module has been created to simplify the import of common modules into unit tests. All you need to do is:

```rs
mod common;
use common::*;
```

✅ **adding a lib.rs for test setups**: this is mainly to add test traits to structures in the future if we need them. It's empty at the moment, I'm just sharing a potential organization.

Of course, all the test files have been updated accordingly (apart from `test/tests/test_estimate_fees.rs`, which I've yet to refactor).

### CODE REVIEW

File: unit_tests/tests/test_block_hash_and_number.rs [hunk 1]
```diff
@@ -1,22 +1,16 @@
...
+use common::*;
 
-use rpc_test::test_config::TestConfig;
-use starknet::{providers::{
-    jsonrpc::{HttpTransport, JsonRpcClient},
-    Provider,
-}, core::types::BlockId};
+use std::collections::HashMap;
+
+use starknet::{providers::{jsonrpc::{HttpTransport, JsonRpcClient}, Provider}, core::types::BlockId};
```
Comment (antiyro) [inline unit_tests/tests/test_block_hash_and_number.rs:6]: rename into starknet_providers

File: unit_tests/src/fixtures.rs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    };
+}
+
+#[fixture]
+pub fn clients() -> HashMap<String, JsonRpcClient<HttpTransport>> {
+    let config = TestConfig::new("./secret.json").expect("'./secret.json' must contain correct node urls");
+    let deoxys = JsonRpcClient::new(HttpTransport::new(
+        Url::parse(&config.deoxys).expect("Error parsing Deoxys node url")
+    ));
+    let alchemy = JsonRpcClient::new(HttpTransport::new(
```
Comment (antiyro) [inline unit_tests/src/fixtures.rs:31]: rename into pathfinder everywhere in code base

File: unit_tests/tests/test_block_number.rs [hunk 3]
```diff
@@ -0,0 +1,18 @@
+mod common;
+use common::*;
+
+use std::collections::HashMap;
+
+use starknet::providers::{jsonrpc::{HttpTransport, JsonRpcClient}, Provider};
```
Comment (antiyro) [inline unit_tests/tests/test_block_number.rs:6]: same

File: unit_tests/tests/test_block_hash_and_number.rs (hunk 1 above)
Comment (Trantorian1) [reply unit_tests/tests/test_block_hash_and_number.rs]: Split imports in commit 56888676622c053e9995b56970f9b7e4542f5e53

File: unit_tests/src/fixtures.rs (hunk 2 above)
Comment (Trantorian1) [reply unit_tests/src/fixtures.rs]: Renamed all 'alchemy' occurrences to 'pathfinder' in commit 185e58ed0bc13217e543784408673836807e87d2. This include `secret.json`, so make sure to update you key names in there!

File: unit_tests/tests/test_block_number.rs (hunk 3 above)
Comment (Trantorian1) [reply unit_tests/tests/test_block_number.rs]: Renamed all 'alchemy' occurrences to 'pathfinder' in commit 185e58ed0bc13217e543784408673836807e87d2. This include `secret.json`, so make sure to update you key names in there!

**Pull request merged**